specs/_features/ePBS/p2p-interface.md

```diff
@@ -0,0 +1,119 @@
+
+In addition to the gossip validations for this topic from prior specifications, the following validations MUST pass before forwarding the `signed_beacon_block` on the network. Alias `block = signed_beacon_block.message`, `signed_bid = block.body.signed_builder_bid`, `bid = signed_bid.message`, `header = bid.header`.
+
+
+New validation:
+
+- _[REJECT]_ The block's execution header timestamp is correct with respect to the slot -- i.e. `header.timestamp == compute_timestamp_at_slot(state, block.slot)`.
+- _[REJECT]_ The bid pubkey is a valid builder in state.
+- _[REJECT]_ The builder has sufficient balances to pay for the bid.
+- _[REJECT]_ The builder signature, `signed_bid.signature`, is valid with respect to the `bid.pubkey`.
+
+###### `execution_payload`
+
+This topic is used to propagate execution payload.
+
+The following validations MUST pass before forwarding the `execution_payload` on the network, assuming the alias `payload = execution_payload`:
+
+- _[IGNORE]_ The payload aligns with header that received in block.
+
+###### `execution_attestation`
+
+This topic is used to propagate signed execution attestation.
+
+The following validations MUST pass before forwarding the `execution_attestation` on the network, assuming the alias `data = execution_attestation.data`:
+
+- _[IGNORE]_ `data.slot` is within the last `ATTESTATION_PROPAGATION_SLOT_RANGE` slots (within a MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance) -- i.e. `data.slot + ATTESTATION_PROPAGATION_SLOT_RANGE >= current_slot >= data.slot` (a client MAY queue future attestations for processing at the appropriate slot).
+- _[REJECT]_ The validator index is within the execution committee in `get_execution_committee(state, data.slot)`
+- _[IGNORE]_ The execution payload being voted for (`data.execution_hash`) has been seen (via both gossip and non-gossip sources) (a client MAY queue execution attestations for processing once payload is retrieved).
+- _[REJECT]_ The signature of `execution_attestation` is valid.
+
+###### `builder_bid`
+
+This topic is used to propagate signed builder bids.
+
+The following validations MUST pass before forwarding the `signed_builder_bid` on the network, assuming the alias `bid = signed_builder_bid.message`:
+
+- _[REJECT]_ The signed builder bid pubkey, `bid.pubkey`, exists in state.
+- _[IGNORE]_ The signed builder bid value, `bid.value`, is less than builder's balance in state.
+- _[IGNORE]_ The signed builder header timestamp is correct with respect to next slot -- i.e. `header.timestamp == compute_timestamp_at_slot(state, current_slot + 1)`.
+- _[IGNORE]_ The signed builder header parent block has matches one of the chain tip in the fork choice store. Builder may submit multiple bids with respect to forks.
+- _[REJECT]_ The builder signature, `signed_bid.signature`, is valid with respect to the `bid.pubkey`.
```

Comment:
each builder can publish an unbounded number of bids. How can we bound it? I would start by requiring increasing bids amounts to be at least % higher than previous

Reply:
I am tempted to say we only allow one bid per builder per fork

Reply:
+1 reject bids on same fork

Reply:
wow pretty interesting visualization, i guess re-bids with at least 10% increase would makes sense similar to tx acceptance